packages/website/blog/2025-01-13-catching-any.mdx

```diff
@@ -0,0 +1,313 @@
+
+There is no `throws` keyword or other way to enforce types of exceptions in TypeScript[^no-throws], so their types can only be declared as `any` or `unknown`.
+Caught exceptions in `catch` blocks and `Promise` rejections are, by default, typed as `any` in TypeScript.
+TypeScript's [`useUnknownInCatchVariables`](https://www.typescriptlang.org/tsconfig/#useUnknownInCatchVariables) changes `catch` block exceptions to `unknown`, but no compiler option equivalent exists for `Promise` rejections.
+
+[`@typescript-eslint/use-unknown-in-catch-callback-variable`](/rules/use-unknown-in-catch-callback-variable) enforces always using the unknown type for the parameter of a Promise rejection callback.
+
+For example, given the following code, TypeScript would not report a type error, but the lint rule would report:
+
+```ts
+function rejectWith(value: string) {
+  return Promise.reject(value);
+}
+
+rejectWith('Nandor').catch(error => {
+  //                       ~~~~~
+  // eslint(@typescript-eslint/use-unknown-in-catch-callback-variable):
+  // Prefer the safe `: unknown` for a `catch` callback variable.
+
+  console.log(error.message); // Should be a type error, but isn't
+});
+```
+
+Had the `error` been given a `: unknown` type, TypeScript would be able to report on `error.message` as being unsafe.
+
+## Banning Usage of Unsafe Types
+
+Once an `any` type exists in code, it is "infectious": it can turn the types of values based on it into more `any`s.
+
+typescript-eslint includes a collection of rules that flag code patterns which spread or are generally made less safe due to an `any`.
+Each of the following rules reports on a specific use of `any`.
+
+- [`@typescript-eslint/no-unsafe-argument`](/rules/no-unsafe-argument) reports on passing an `any` typed value to a function call:
+
+  ```ts
+  declare function greet(friend: string): void;
+
+  function greetFromData(raw: string) {
+    greet(JSON.parse(raw));
+    //    ~~~~~~~~~~~~~~~
+    // eslint(@typescript-eslint/no-unsafe-argument):
+    // Unsafe argument of type `any` assigned to a parameter of type `string`.
+  }
+  ```
+
+* [`@typescript-eslint/no-unsafe-assignment`](/rules/no-unsafe-assignment) reports on assigning an `any` typed value to a property or variable:
+
+  ```ts
+  const person = {
+    name: '(anonymous)',
+  };
+
+  function setNameFromData(raw: string) {
+    person.name = JSON.parse(raw);
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // eslint(@typescript-eslint/no-unsafe-argument):
+    // Unsafe assignment of an `any` value.
+  }
+  ```
+
+* [`@typescript-eslint/no-unsafe-call`](/rules/no-unsafe-call) reports on calling an `any` typed value like a function:
+
+  ```ts
+  const person = {
+    name: '(anonymous)',
+  };
+
+  function setNameFromData(raw: string) {
+    person.name = JSON.parse(raw);
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // eslint(@typescript-eslint/no-unsafe-argument):
+    // Unsafe assignment of an `any` value.
+  }
+  ```
```

Comment:
I think this example accidentally made it in (it's an example for `no-unsafe-assignment` rather than `no-unsafe-call`).